README.md

```diff
@@ -16,33 +14,76 @@ go install github.com/bwplotka/bingo@latest
-
-## Custom templates
+### How to use custom templates
 
 In order to generate the client, `go-swagger` uses default templates. These templates can be customised to add custom configuration that are applied each time the client is generated.
 
 For more information, check out the `go-swagger` docs on how to [use custom templates](https://github.com/go-swagger/go-swagger/blob/master/docs/generate/templates.md). The default template definitions for the client can be found in [go-swagger/generator/templates/client/](https://github.com/go-swagger/go-swagger/tree/master/generator/templates/client).
 
 In this project, the custom templates can be found in `templates/`. They are provided to the generation command through the flag `--template-dir=templates`.
 
-The custom templates provide added functionality for things such as authentication, TLS/SSL, retries, and custom error handling.
\ No newline at end of file
+The custom templates provide added functionality for things such as authentication, TLS/SSL, retries, and custom error handling.
+
+## Build the client
+
+### Configuration
+
+The client has the following friendly configuration options:
+
+```go
+import goapi "github.com/grafana/grafana-openapi-client-go/client"
+
+cfg := &goapi.TransportConfig{
+    // Host is the doman name or IP address of the host that serves the API.
+    Host:       "localhost:3000",
+    // BasePath is the URL prefix for all API paths, relative to the host root.
+    BasePath:   "/api",
+    // Schemes are the transfer protocols used by the API (http or https).
+    Schemes:    []string{"http"},
+    // APIKey is an optional API key or service account token.
+    APIKey:     os.Getenv("API_ACCESS_TOKEN"),
+    // BasicAuth is optional basic auth credentials.
+    BasicAuth:  url.UserPassword("admin", "admin"),
+    // OrgID provides an optional organization ID.
+    // OrgID is only supported with BasicAuth since API keys are already org-scoped.
+    OrgID:      1,
+    // TLSConfig provides an optional configuration for a TLS client
+    TLSConfig:  &tls.Config{},
+    // NumRetries contains the optional number of attempted retries
+    NumRetries: 3,
+    // RetryTimeout sets an optional time to wait before retrying a request
+    RetryTimeout: 0,
+    // RetryStatusCodes contains the optional list of status codes to retry
+    // Use "x" as a wildcard for a single digit (default: [429, 5xx])
+    RetryStatusCodes []string{"420, 5xx"},
+}
+
+client := goapi.NewHTTPClientWithConfig(strfmt.Default, cfg)
+```
+
+### Examples
+
+Checkout how the Grafana Terraform Provider initialises and uses the client [here](https://github.com/grafana/terraform-provider-grafana/blob/master/internal/provider/provider.go#L411).
```

Comment:
nit: better use a [permanent link](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-a-permanent-link-to-a-code-snippet)